rs/tests/ckbtc/src/adapter.rs

```diff
@@ -23,13 +23,14 @@ use std::{str::FromStr, time::Duration};
 /// This allows to make arbitrary calls to the adapter, but they will go through the
 /// entire replica code path.
 #[derive(Clone)]
-pub struct AdapterProxy<'a> {
+pub struct AdapterProxy<'a, T: IcRpcClientType> {
+    _network: T,
     msg_can: MessageCanister<'a>,
     log: Logger,
 }
 
-impl<'a> AdapterProxy<'a> {
-    pub async fn new(agent: &'a Agent, log: Logger) -> Self {
+impl<'a, T: IcRpcClientType> AdapterProxy<'a, T> {
+    pub async fn new(_network: T, agent: &'a Agent, log: Logger) -> Self {
```

Comment:
Are planning to have different instanciations of the type?
Otherwise, it is not necessary, we could leave this out and initialize them via `AdapterProxy::<'_, BitcoinNetwork>::new(...)`